aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,800 @@
+                    read_dims=this_chunk_read_dims,
+                ),
+                shape=sample_chunk_shape,
+                dtype=pixel_type,
+            )
+
+        # Convert the numpy array of lazy readers into a dask array and fill the inner
+        # most empty dimensions with chunks
+        merged = da.block(lazy_arrays.tolist())
+
+        # Because we have set certain dimensions to be chunked and others not
+        # we will need to transpose back to original dimension ordering
+        # Example being, if the original dimension ordering was "SZYX" and we want to
+        # chunk by "S", "Y", and "X" we created an array with dimensions ordering "ZSYX"
+        transpose_indices = []
+        transpose_required = False
+        for i, d in enumerate(dims_str):
+            new_index = blocked_dimension_order.index(d)
+            if new_index != i:
+                transpose_required = True
+                transpose_indices.append(new_index)
+            else:
+                transpose_indices.append(i)
+
+        # Only run if the transpose is actually required
+        # The default case is "Z", "Y", "X", which _usually_ doesn't need to be
+        # transposed because that is _usually_ the normal dimension order of the CZI
+        # file anyway
+        if transpose_required:
+            merged = da.transpose(merged, tuple(transpose_indices))
+
+        # Because dimensions outside of Y and X can be in any order and present or not
+        # we also return the dimension order string.
+        return merged
+
+    @staticmethod
+    def _get_coords_and_physical_px_sizes(
+        xml: ET.Element, scene_index: int, dims_shape: Dict[str, Any]
+    ) -> Tuple[Dict[str, Any], types.PhysicalPixelSizes]:
+        # Create coord dict
+        coords = {}
+
+        # Get all images
+        img_sets = xml.findall(".//Image/Dimensions/Channels")
+
+        if len(img_sets) != 0:
+            # Select the current scene
+            img = img_sets[0]
+            if scene_index < len(img_sets):
+                img = img_sets[scene_index]
+
+            # Construct channel list
+            scene_channel_list = []
+            channels = img.findall("./Channel")
+            for channel in channels:
+                channel_name = channel.attrib["Name"]
+                scene_channel_list.append(channel_name)
+
+            # Attach channel names to coords
+            coords[DimensionNames.Channel] = scene_channel_list
+
+        # Unpack short info scales
+        list_xs = xml.findall(".//Distance[@Id='X']")
+        list_ys = xml.findall(".//Distance[@Id='Y']")
+        list_zs = xml.findall(".//Distance[@Id='Z']")
+        scale_xe = list_xs[0].find("./Value")
+        scale_ye = list_ys[0].find("./Value")
+        scale_ze = None if len(list_zs) == 0 else list_zs[0].find("./Value")
+
+        # Unpack the string value to a float
+        # Split by "E" and take the first part because the values are stored
```

Comment:
Should `"E"` also be a file constant?

Reply:
I could go either way. I think I side slightly on not because it's only used in this one place.